File: step-function-map-in-map/sample_data_sets/nested_generate.go
package main

import (
        "fmt"
        "encoding/json"
        "os"
        "log"
)

type Request struct {
        CaseId string `json:"caseId"`
}

type RequestSlot struct {
        Requests []Request `json:"requests"`
}

type RequestMessage struct {
        RequestId string `json:"requestId"`
        RequestSlots []RequestSlot `json:"requestSlots"`
}

func main() {
        targets := []int{1, 5, 10, 20, 40, 100, 200, 500, 700, 1000}
        limit := 40

        for _,target := range targets {
                // Find the x slots (num of arrays)
                slot := target / limit

                // Remaining divdend will go to a new slot
                if (target % limit) > 0 {
                        slot++
                }

                requestSlots := make([]RequestSlot, 0)

                counter := 0; 

                for targetSlot := 0; targetSlot < slot; targetSlot++ {
                        requests := make([]Request, 0)
                        requestSlot := RequestSlot{requests}
                        // Inject current slot into slot array
                        requestSlots = append(requestSlots, requestSlot)
                
                        // Get this slot limit
                        slotatarget :=  (targetSlot + 1) *  limit

                        // If exceed the total target, then will be final target
                        if slotatarget > target {
                                slotatarget = target
                        }

                        // Loop through all numbers for current slot
                        for ; counter < slotatarget; counter++ {
                                requestSlots[targetSlot].Requests = append(requestSlots[targetSlot].Requests, Request{fmt.Sprintf("%d", counter)})
                        }
                }



                m := RequestMessage{"123456", requestSlots}

                b, _ := json.Marshal(m)

                fmt.Println("")
                fmt.Println(fmt.Sprintf("Sample Dataset for %d cases:", target))
                fmt.Println(string(b))

		err := os.WriteFile(fmt.Sprintf("dataset_%d.json", target), []byte(string(b)), 0644)
		if err != nil {
				log.Fatal(err)
		}
        }
}

File: step-function-map-in-map/sample_data_sets/nested_generate.js
#!/usr/bin/env node

'use strict';

const fs = require('fs');
const util = require('util');

class Request {
    constructor(caseId) {
        this.caseId = caseId;
    }
}

class RequestSlot {
    constructor(requests) {
        this.requests = requests;
    }
}

class RequestMessage {
    constructor(requestId, requestSlots) {
        this.requestId = requestId;
        this.requestSlots = requestSlots;
    }
}

function main() {
    var targets = [1, 5, 10, 20, 40, 100, 200, 500, 700, 1000];
    var limit = 40;

    targets.forEach(function (target) {

        // Find the x slots (num of arrays)
        // Need to use integer to not get decimal
        var slot = parseInt(target / limit);

        // Remaining divdend will go to a new slot
        if ((target % limit) > 0) {
            slot++
        }

        var requestSlots = [];

        var counter = 0;


        for (var targetSlot = 0; targetSlot < slot; targetSlot++) {
            var requests = []
            var requestSlot = new RequestSlot(requests);

            // Inject current slot into slot array
            requestSlots.push(requestSlot);

            // Get this slot limit
            var slotatarget = (targetSlot + 1) * limit;

            // If exceed the total target, then will be final target
            if (slotatarget > target) {
                slotatarget = target
            }

            // Loop through all numbers for current slot
            while (counter < slotatarget) {
                requestSlots[targetSlot].requests.push(new Request(util.format("%d", counter)));
                counter++
            }
        }

        var requestMessage = new RequestMessage('123456', requestSlots);

        var content = JSON.stringify(requestMessage);

        console.log("");
        console.log(util.format("Sample Dataset for %d cases:", target));
        console.log(content);

        try {
            fs.writeFileSync(util.format('dataset_%d.json', target), content);
            //file written successfully
        } catch (err) {
            console.error(err)
        }
    });
}

main();